### ISSUE TITLE
Add the ability to pass an `--overlay` option to singularity/apptainer

### ISSUE THREAD

**Issue by DrDaveD** [glideinWMS/glideinwms#229]
**Is your feature request related to a problem? Please describe.**
Annie users made a Fermilab Service Now ticket (RITM1316133) expressing the desire to more easily overlay developer-specific software on top of their standard container, and to pass it in to jobs.  

**Describe the solution you'd like**
It would be good to have a job submission option to specify the path to an overlay path to pass to the `--overlay` option in singularity or apptainer.  It will most likely have a `:ro` on the end to indicate it as read-only, but that should just be passed through.  For security reasons it probably should be limited to paths that are in the job's own temporary space or under /cvmfs (the latter for the Rapid Code Distribution System).

**Describe alternatives you've considered**
I was told that currently people avoid using the container at all because they have to mess with lots of bind mounts and symlinks if they want to do something like this.

**Info (please complete the following information):**

-   Priority: Priority level for this feature [medium]
-   Stakeholder: Concerned stakeholder(s) [FIFE]
-   Component: The affected component, if any, from this feature [frontend]

**Additional context**

**Comment by DrDaveD** [comment]
Oh and I believe that RCDS puts a symlink in the job's own temporary space so it may not need to worry about allowing /cvmfs.

**Comment by mambelli** [comment]
As a workaround, the VO can put any arbitrary option in the GLIDEIN_SINGULARITY_OPTS attribute (in the "attrs" section of the Frontend configuration). This is a string that is added to the command line invocation of Singularity. It can be limited to a set of jobs if it is in a group configuration.

**Comment by mambelli** [comment]
@DrDaveD Is this request to allow the jobs to specify arbitrary additional Singularity options? Or is preferable for the option to be just the strings for the `--overlay` option?
Especially if it is the latter, which, if any, checks should GWMS do on the string?

**Comment by DrDaveD** [comment]
I don't think we want job submitters to be allowed to specify arbitrary options because they would be able to use that to interfere with jobs from other people on the same machine.  As I put in the description, I think the option should be limited to paths within the job's own temporary space, and optionally with a ":ro" on the end.
